crates/virtio-queue/src/queue.rs

```diff
@@ -555,8 +582,19 @@ mod tests {
             i += 1;
             q.disable_notification().unwrap();
 
-            while let Some(_chain) = q.iter().unwrap().next() {
-                // In a real use case, we would do something with the chain here.
+            while let Some(chain) = q.iter().unwrap().next() {
+                // Process the descriptor chain, and then add entries to the
+                // used ring.
+                let head_index = chain.head_index();
+                let mut desc_len = 0;
+                chain.for_each(|d| {
+                    if d.flags() & VIRTQ_DESC_F_WRITE == VIRTQ_DESC_F_WRITE {
+                        desc_len += d.len();
+                    }
+                });
```

Comment:
We can probably update every for_each with the way it is here so we don't have to care if the descriptors' flags have a particular value.

Reply:
If I do so, I'll have to set in each test a few descriptors with the flag `VIRTQ_DESC_F_WRITE` otherwise the `add_used()` won't be called, and this will change the expected used index. WDYT?

Reply:
But if you parse a chain and there are no writable descriptors, I think you still have to add an used entry, even if with len = 0.

Reply:
Actually I think you're correct, let me update the tests :)

Reply:
Done!